packages/ra-data-simple-rest/src/index.spec.ts

```diff
@@ -68,6 +68,65 @@ describe('Data Simple REST Client', () => {
             expect(result.total).toEqual(42);
         });
     });
+    describe('getOne', () => {
+        it('should allow numeric id in path', async () => {
+            const httpClient = jest.fn().mockResolvedValue({ id: 'Post#123' });
+            const client = simpleClient('http://localhost:3000', httpClient);
+
+            await client.getOne('posts', { id: 123 });
+
+            expect(httpClient).toHaveBeenCalledWith(
+                'http://localhost:3000/posts/123',
+                expect.any(Object)
+            );
+        });
+        it('should escape id in path', async () => {
+            const httpClient = jest.fn().mockResolvedValue({ id: 'Post#123' });
+            const client = simpleClient('http://localhost:3000', httpClient);
+
+            await client.getOne('posts', { id: 'Post#123' });
+
+            expect(httpClient).toHaveBeenCalledWith(
+                'http://localhost:3000/posts/Post%23123',
+                expect.any(Object)
+            );
+        });
+    });
+    describe('update', () => {
+        it('should escape id in path', async () => {
+            const httpClient = jest.fn().mockResolvedValue({ id: 'Post#123' });
+            const client = simpleClient('http://localhost:3000', httpClient);
+
+            await client.update('posts', {
+                previousData: undefined,
+                id: 'Post#123',
+                data: { body: '' },
+            });
+
+            expect(httpClient).toHaveBeenCalledWith(
+                'http://localhost:3000/posts/Post%23123',
+                expect.any(Object)
+            );
+        });
+    });
+    describe('updateMany', () => {
+        it('should escape id in path', async () => {
+            const httpClient = jest
+                .fn()
+                .mockResolvedValue({ json: { id: 'Post#123' } });
```

Comment:
This isn't a valid `updateMany` response and it will cause warnings.

Reply:
Fixing.